draft-ietf-quic-http.md

```diff
@@ -591,6 +591,83 @@ The payload consists of:
   : HPACK-compressed request headers for the promised response.
 
 
+### GOAWAY {#frame-goaway}
+
+The GOAWAY frame (type=0x7) is used to initiate graceful shutdown of a
+connection by a server.  GOAWAY allows a server to stop accepting new requests
+while still finishing processing of previously received requests.  This enables
+administrative actions, like server maintenance.  GOAWAY by itself does not
+close a connection.
+
+The GOAWAY frame does not define any flags, and the payload is a QUIC stream
+identifier.  The GOAWAY frame applies to the connection, not a specific stream.
+An endpoint MUST treat a GOAWAY frame on a stream other than the control stream
+as a connection error ({{errors}}) of type HTTP_WRONG_STREAM.
+
+There is an inherent race condition between a client sending new requests and
+the server sending a GOAWAY frame.  To deal with this case, the GOAWAY contains
```

Comment:
We never explained the nature of that race condition in 7540, and now it's bugging me.  Should we open an issue to track this?

Maybe the right way to fix this is to change the description.  Say that this frame exists to allow client and server to agree on what requests were accepted and which were rejected.  GOAWAY identifies the last request that the client can make and the server will accept.  All requests that are sent on higher-numbered streams are implicitly cancelled.

Described that way, there isn't a need to talk about the race condition.